odoo/addons/test_testing_utilities/tests/test_xml_tools.py

```diff
@@ -0,0 +1,153 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+
+from lxml import etree
+from odoo.tests import common
+from odoo.tools.xml_utils import cleanup_xml_node
+
+
+class TestXMLTools(common.TransactionCase):
+
+    def setUp(self):
+        super(TestXMLTools, self).setUp()
+        self.qweb_poor = self.env()['ir.ui.view'].create({
+            'type': 'qweb',
+            'arch_db': """
+    <h1>
+            <h2/>
+                <h2>text</h2>
+        \t<h2><h3/></h2>
+            <h2>            </h2>
+<!-- comment removed by qweb -->
+</h1>"""})
+
+    def test_cleanup_xml_node_no_modif(self):
+        # Qweb removes comments and any whitespace before first tag, no other content affected
+        expected = """<h1>
+            <h2/>
+                <h2>text</h2>
+        \t<h2><h3/></h2>
+            <h2>            </h2>
+
```

Comment:
@odoostan I added this white-line, do you have an opinion on this?
I couldn't quickly find which commit is adding it in the qweb rendering

Reply:
Thanks, that's what I would have done as well.
No idea what affected the qweb rendering, I couldn't find it either :/